src/transaction/log_global.c

```diff
@@ -103,4 +103,18 @@ log_global::~log_global ()
 {
   delete writer_info;
 }
+
+void
+log_global::update_max_ps_flushed_lsa (const LOG_LSA &lsa)
+{
+  m_max_ps_flushed_lsa = lsa;
+  m_ps_lsa_cv.notify_all ();
+}
+
+void
+log_global::wait_flushed_lsa (const log_lsa &flush_lsa)
+{
+  std::unique_lock<std::mutex> lock (m_ps_lsa_mutex);
```

Comment:
Is it ok to add a small optimization here, to check the condition first before locking:
```suggestion
  if (m_max_ps_flushed_lsa >= flush_lsa)
    {
      // already flushed
      return;
    }
  std::unique_lock<std::mutex> lock (m_ps_lsa_mutex);
```